lib/taglet/tag_as.ex

```diff
@@ -45,4 +49,32 @@ defmodule Taglet.TagAs do
       end
     end
   end
+
+  defmacro __before_compile__(_env) do
+
+    quote do
+      @contexts
+      |> Enum.each(fn(context) ->
+        Module.eval_quoted(__MODULE__, quote do
+          def unquote(:"add_#{context}")(tags) do
+            Taglet.add(%__MODULE__{}, tags, unquote(context))
+          end
+
+          def unquote(:"add_#{Inflex.singularize(context)}")(tags) do
```

Comment:
What do you mean? To use the `singularized_context ` declared before? I can't but I could do something like this:
```elixir
@contexts
  |> Enum.each(fn(context) ->
    singularized_context = Inflex.singularize(context)

    def unquote(:"add_#{singularized_context}")(tags) do
      Taglet.add(%__MODULE__{}, tags, unquote(context))
     end

  end)

``` 

In order to execute the Inflex only once.